python/lsst/ap/verify/testPipeline.py

```diff
@@ -149,10 +148,20 @@ def run(self, ccdExposure, *, camera=None, bias=None, linearizer=None,
                       )
 
 
+class MockCharacterizeImageConfig(CharacterizeImageConfig):
+
+    def setDefaults(self):
+        super().setDefaults()
+        # Avoid delegating to lsst.obs.base.Instrument specialization for the
+        # data ID packing algorithm to use, since test code often does not use a
+        # real Instrument in its data IDs.
+        self.idGenerator.packer.name = "observation"
```

Comment:
I'm confused -- why can't you just use `CharacterizeImageConfig` as-is?

Since [some of the tests assume that the correct connections are drawn from the "real" tasks](https://github.com/lsst/ap_verify/blob/main/tests/test_testPipeline.py#L48-L49), creating custom configs is a dangerous precedent even if they don't fiddle with the connections themselves.

Reply:
That might be a good idea. The test suite that actually _uses_ `MockApPipe.yaml`, `test_driver.py`, is based on [ap_verify_testdata](https://github.com/lsst/ap_verify_testdata), which uses `LSSTCam-imSim` as its instrument.

Reply:
`obs_base` has a `Instrument` specifically designed for testing if that helps at all.

Reply:
Ah, it turns out `test_driver.py` was already fine and it was `test_testPipeline.py` that was all that was giving me trouble.  And @timj 's suggestion of `lsst.obse.base.instrument_tests.DummyCam` is the perfect solution for that.